code/controllers/subsystems/hallucinations.dm

```diff
@@ -29,6 +29,9 @@
 	var/list/candidates = list()
 	for(var/T in all_hallucinations)
 		var/datum/hallucination/H = new T
+		if(isnull(H))
+			log_debug("Null hallucination found while creating new in subsystem.")
+			continue
```

Comment:
This seems like a weird thing to add? If you're using subtypesof you should like never get a null hallucination out of that. Something is weird elsewhere.

Reply:
Thinking on it, yeah. The isnull() in handle_hallucination should be enough. I'll remove it when I can.